cloud/blockstore/libs/storage/volume/partition_requests.h

```diff
@@ -865,6 +867,107 @@ class TPartitionRequestActor final
         }
     }
 
+    void Merge(
+        NProto::TDescribeBlocksResponse& src,
+        ui32 requestNo,
+        NProto::TDescribeBlocksResponse& dst)
+    {   
+        if (FAILED(src.GetError().GetCode())) {
+            *dst.MutableError() = std::move(*src.MutableError());
+        } else {
+            for (const auto& freshBlockRange: src.GetFreshBlockRanges()) {
+                SplitFreshBlockRangeFromRelativeToGlobalIndices(
+                    freshBlockRange,
+                    requestNo,
+                    &dst);
+            }
+
+            const auto& srcBlobPieces = src.GetBlobPieces();
+
+            for (const auto& blobPiece: srcBlobPieces) {
+                NProto::TBlobPiece dstBlobPiece;
+                dstBlobPiece.MutableBlobId()->CopyFrom(blobPiece.GetBlobId());
+                dstBlobPiece.SetBSGroupId(blobPiece.GetBSGroupId());
+
+                for (const auto &srcRange : blobPiece.GetRanges()) {
+                    SplitBlobPieceRangeFromRelativeToGlobalIndices(
+                        srcRange,
+                        requestNo, 
+                        &dstBlobPiece);
+                }
+                dst.MutableBlobPieces()->Add(std::move(dstBlobPiece));
+            }    
+        }
+    }
+
+    void SplitFreshBlockRangeFromRelativeToGlobalIndices(
+        const NProto::TFreshBlockRange& srcRange,
+        ui32 requestNo,
+        NProto::TDescribeBlocksResponse* dst)
+    {
+        const ui32 startIndex = srcRange.GetStartIndex();
+        ui32 blocksCount = 0;
+
+        while (blocksCount < srcRange.GetBlocksCount()) {
+            const auto index = RelativeToGlobalIndex(
+                BlocksPerStripe,
+                startIndex + blocksCount,
+                PartitionsCount,
+                PartitionRequests[requestNo].PartitionId);
+
+            const auto stripeRange = GetStripeRange(BlocksPerStripe, index);
+
+            const auto rangeBlocksCount = std::min(
+                static_cast<ui64>(stripeRange.End) - index + 1,
+                static_cast<ui64>(srcRange.GetBlocksCount()) - blocksCount);
+
+            NProto::TFreshBlockRange dstRange;
+            dstRange.SetStartIndex(index);
+            dstRange.SetBlocksCount(rangeBlocksCount);
+            const ui64 bytesCount = static_cast<ui64>(rangeBlocksCount) * BlockSize;
+            for (ui64 i = 0; i < bytesCount; i++) {
+                dstRange.MutableBlocksContent()->append(srcRange.GetBlocksContent()[i]);
+            }
+
+            blocksCount += rangeBlocksCount;
+
+            dst->MutableFreshBlockRanges()->Add(std::move(dstRange));
+        }
+    }
+
+    void SplitBlobPieceRangeFromRelativeToGlobalIndices(
```

Comment:
Can we generalize this with SplitFreshBlockRangeFromRelativeToGlobalIndices?

Reply:
we can't avoid the implementation of these two functions

if we add templates code will be mode difficult